iothub/device/samples/convention-based-samples/TemperatureController/TemperatureControllerSample.cs

```diff
@@ -0,0 +1,375 @@
+            _maxTemp[Thermostat2] = 0d;
+
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                if (temperatureReset)
+                {
+                    // Generate a random value between 5.0°C and 45.0°C for the current temperature reading for each "Thermostat" component.
+                    _temperature[Thermostat1] = Math.Round(s_random.NextDouble() * 40.0 + 5.0, 1);
+                    _temperature[Thermostat2] = Math.Round(s_random.NextDouble() * 40.0 + 5.0, 1);
+                }
+
+                await SendTemperatureAsync(Thermostat1, cancellationToken);
+                await SendTemperatureAsync(Thermostat2, cancellationToken);
+                await SendDeviceMemoryAsync(cancellationToken);
+
+                temperatureReset = _temperature[Thermostat1] == 0 && _temperature[Thermostat2] == 0;
+                await Task.Delay(5 * 1000);
+            }
+        }
+
+        // The callback to handle property update requests.
+        private async Task HandlePropertyUpdatesAsync(ClientPropertyCollection writableProperties, object userContext)
+        {
+            foreach (KeyValuePair<string, object> writableProperty in writableProperties)
+            {
+                switch (writableProperty.Key)
+                {
+                    case Thermostat1:
+                    case Thermostat2:
+                        const string targetTemperatureProperty = "targetTemperature";
+                        if (writableProperties.TryGetValue(writableProperty.Key, targetTemperatureProperty, out double targetTemperatureRequested))
+                        {
+                            await HandleTargetTemperatureUpdateRequestAsync(writableProperty.Key, targetTemperatureRequested, writableProperties.Version, userContext);
+                            break;
+                        }
+                        else
+                        {
+                            _logger.LogWarning($"Property: Received an unrecognized property update from service for component {writableProperty.Key}:" +
+                                        $"\n{{ {writableProperty.Value} }}.");
+                            break;
+                        }
+
+                    default:
+                        _logger.LogWarning($"Property: Received an unrecognized property update from service:" +
+                                        $"\n{{ {writableProperty.Key}: {writableProperty.Value} }}.");
+                        break;
+                }
+            }
+        }
+
+        private async Task HandleTargetTemperatureUpdateRequestAsync(string componentName, double targetTemperature, long version, object userContext)
+        {
+            const string targetTemperatureProperty = "targetTemperature";
+            _logger.LogDebug($"Property: Received - component=\"{componentName}\", {{ \"{targetTemperatureProperty}\": {targetTemperature}°C }}.");
+
+            // Update Temperature in 2 steps
```

Comment:
I think a more accurate comment is to update the temperature on both components

Reply:
What I mean to convey is: this will simulate the temperature update in two steps; i.e. if the current temp is 10 and the desired is 30, it'll go 10 (current) => 20 => 30 (desired).  It'll still happen on the single component on which it was requested though.

Reply:
I'll add an example to describe what I mean

Reply:
I was sort of guessing the reason for this 2 step update till I saw this. I'm wondering if this is really needed in a sample? Why not keep it simple and just update it at once? I feel the aim is to show best practices to use the APIs but not guess user scenarios. To me it feels like that logic should be determined by the user based on the use case.

Reply:
sure, I don't mind a one-step update. I just felt a 2-step better simulates a real scenario